src/debug.c

```diff
@@ -1052,6 +1052,46 @@ NULL
 
 /* =========================== Crash handling  ============================== */
 
+/* When hide-user-data-from-log is enabled, to avoid leaking user info, we only
+ * print tokens of the current command into the log. First, we collect command
+ * tokens into this struct (Commands tokens are defined in json schema). Later,
+ * checking each argument against the token list. */
+#define CMD_TOKEN_MAX_COUNT 128 /* Max token count in a command's json schema */
+struct cmdToken {
+    const char *tokens[CMD_TOKEN_MAX_COUNT];
```

Comment:
Maybe we should add simple logic of  token overflow. If `n_token` value is `CMD_TOKEN_MAX_COUNT` then return. And correspondigly on `cmdTokenCheck` if `n_token` value reached the limit then hide all arguments.

Reply:
Currently, we don't even have a command with 16 tokens in redis core. ft.search seems like will have 42 tokens as the largest one. 128 is a quite generous limit. Also, I feel like printing as much as we can is better than hiding arguments. If we have a complaint that we miss some of the arguments, then we will increment the limit. Wdyt?

Reply:
I am afraid that a module might define a vector command with many arguments of 256 or more... Just thinking out loud. 

The overflow logic supposed to be rather simple.

Reply:
Not sure if we are on the same page, you realize that overflow will not actually happen? Please see line 1070, there is a guard in the condition. `n_token` will not be larger than 128.

So, if we get a command with 128+ tokens, with the current approach, we'll take first 128 tokens into account. We may potentially fail to print some tokens if they are not in the first 128 tokens. Your suggestion is not printing any of the tokens if command has more than 128 tokens?

Reply:
Missed that condition. Thanks.